### PR TITLE
Feat/ec 4368 stickynav desktop directive

### PR DESCRIPTION
**Pull request by lbiedinger**

### CODE REVIEW

File: plugins/vue-directives.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  inserted: (el) => {
+    el.scrolledVisible = true;
+    window.addEventListener('scroll', () => handleScroll(el));
+    window.addEventListener('orientationchange', () => handleOrientationChange(el));
+  }
+});
+
+const handleScroll = (el) => {
+  const newPosition = window.scrollY;
+  if (onDesktop && el.scrollPosition < newPosition && el.scrollPosition > 150 && el.scrolledVisible) {
```
Comment (monkemedia) [inline plugins/vue-directives.js:13]: 150 is when the scroll kicks in? If so, wonder if we should pass this value from the vue component?

File: plugins/vue-directives.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    el.scrolledVisible = false;
+  } else if (onDesktop && ((el.scrollPosition - 5) > newPosition || el.scrollPosition <= 150) && !el.scrolledVisible) {
+    el.setAttribute('style', 'transform: translate3d(0, 0, 0)');
+    el.scrolledVisible = true;
+  }
+  el.scrollPosition = newPosition;
+};
+
+const handleOrientationChange = (el) => {
+  console.log('orientation');
```
Comment (monkemedia) [inline plugins/vue-directives.js:24]: Do we need this console log?

File: plugins/vue-directives.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import Vue from 'vue';
+
+Vue.directive('visible-on-scroll', {
+  inserted: (el) => {
+    el.scrolledVisible = true;
+    window.addEventListener('scroll', () => handleScroll(el));
```
Comment (monkemedia) [inline plugins/vue-directives.js:6]: I can assume there are no issues with window being undefined due to SSR?

File: plugins/vue-directives.js (hunk 2 above)
Comment (lbiedinger) [reply plugins/vue-directives.js:24]: this should be removed

File: plugins/vue-directives.js (hunk 1 above)
Comment (lbiedinger) [reply plugins/vue-directives.js:13]: Potentially, but to limit the scope let's leave it hardcoded for now until we have an actual usecase.

File: plugins/vue-directives.js (hunk 3 above)
Comment (alansutherland) [reply plugins/vue-directives.js]: haven't seen any ssr issues while testing, but just to be safe I've added this conditional so it'll only run client-side

   ```
if (process.browser) {
      window.addEventListener('scroll', () => handleScroll(el));
      window.addEventListener('orientationchange', () => handleOrientationChange(el));
}
```

**Pull request merged**